### PR TITLE
Minor tidy-ups of LGC and its interface

### PR DESCRIPTION
**Pull request by trenouf**
All three commits are part of this PR.

The commits do a few bits of tidying in LGC and its interface, including removing the deprecated NGG fast launch.

### CODE REVIEW

**Comment by JaxLinAMD** [comment]
retest this please

**Comment by trenouf** [comment]
V3: Oops again. Fix lgc tests.

File: llpc/context/llpcPipelineContext.cpp [hunk 1]
```diff
@@ -635,17 +636,35 @@ void PipelineContext::setGraphicsStateInPipeline(Pipeline *pipeline) const {
...
-  inputAssemblyState.topology = static_cast<PrimitiveTopology>(inputIaState.topology);
+  switch (inputIaState.topology) {
+  case VK_PRIMITIVE_TOPOLOGY_POINT_LIST:
+    inputAssemblyState.primitiveType = PrimitiveType::Point;
+    break;
+  case VK_PRIMITIVE_TOPOLOGY_LINE_LIST:
+  case VK_PRIMITIVE_TOPOLOGY_LINE_STRIP:
+  case VK_PRIMITIVE_TOPOLOGY_LINE_LIST_WITH_ADJACENCY:
+  case VK_PRIMITIVE_TOPOLOGY_LINE_STRIP_WITH_ADJACENCY:
+    inputAssemblyState.primitiveType = PrimitiveType::Point;
```
Comment (amdrexu) [inline llpc/context/llpcPipelineContext.cpp:647]: Should this be PrimitiveType::Line?

File: llpc/context/llpcPipelineContext.cpp [hunk 2]
```diff
@@ -635,17 +636,35 @@ void PipelineContext::setGraphicsStateInPipeline(Pipeline *pipeline) const {
...
+    break;
+  case VK_PRIMITIVE_TOPOLOGY_TRIANGLE_LIST:
+  case VK_PRIMITIVE_TOPOLOGY_TRIANGLE_STRIP:
+  case VK_PRIMITIVE_TOPOLOGY_TRIANGLE_FAN:
+  case VK_PRIMITIVE_TOPOLOGY_TRIANGLE_LIST_WITH_ADJACENCY:
+  case VK_PRIMITIVE_TOPOLOGY_TRIANGLE_STRIP_WITH_ADJACENCY:
+    inputAssemblyState.primitiveType = PrimitiveType::Point;
```
Comment (amdrexu) [inline llpc/context/llpcPipelineContext.cpp:654]: PrimitiveType::Triangle?

**Review by amdrexu — COMMENTED** [review]
LGTM

**Comment by trenouf** [comment]
V4: Fix LLPC front-end setting of prim type.

File: llpc/context/llpcPipelineContext.cpp (hunk 1 above)
Comment (trenouf) [reply llpc/context/llpcPipelineContext.cpp]: Yes; thanks.

**Comment by trenouf** [comment]
The jenkins test failed because it failed to git fetch the Khronos headers. I suspect it's running in the Shanghai office and they have been having connectivity issues there. The equivalent internal test run worked fine.

**Pull request merged**